From bd591602d76469377e80dfc42e920fb01e1279ec Mon Sep 17 00:00:00 2001
From: Andy Grover <agrover@redhat.com>
Date: Fri, 5 Sep 2014 16:35:59 -0400
Subject: [infiniband] ulp/isert: Fix wrong connection requests list addition

Message-id: <1409934989-29941-2-git-send-email-agrover@redhat.com>
Patchwork-id: 91462
O-Subject: [RHEL7.1 PATCH e-stor 01/31] Target/iser: Fix wrong connection requests list addition
Bugzilla: 1129387
RH-Acked-by: Chris Leech <cleech@redhat.com>
RH-Acked-by: Rob Evers <revers@redhat.com>

From: Sagi Grimberg <sagig@mellanox.com>

Should be adding list_add_tail($new, $head) and not
the other way around.

Signed-off-by: Sagi Grimberg <sagig@mellanox.com>
Cc: stable@vger.kernel.org # 3.10+
Signed-off-by: Nicholas Bellinger <nab@linux-iscsi.org>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/ulp/isert/ib_isert.c b/drivers/infiniband/ulp/isert/ib_isert.c
index 087fe48..2b95493 100644
--- a/drivers/infiniband/ulp/isert/ib_isert.c
+++ b/drivers/infiniband/ulp/isert/ib_isert.c
@@ -463,7 +463,7 @@ isert_connect_request(struct rdma_cm_id *cma_id, struct rdma_cm_event *event)
   goto out_conn_dev;
 
  mutex_lock(&isert_np->np_accept_mutex);
- list_add_tail(&isert_np->np_accept_list, &isert_conn->conn_accept_node);
+ list_add_tail(&isert_conn->conn_accept_node, &isert_np->np_accept_list);
  mutex_unlock(&isert_np->np_accept_mutex);
 
  pr_debug("isert_connect_request() waking up np_accept_wq: %p\n", np);
-- 
1.7.1